rust/Cargo.toml

```diff
@@ -4,6 +4,7 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
+inventory = "0.3.0"
```

Comment:
We should replace inventory, which involves "life-before-main", by linkme which rely solely on linker section and do all its magic during compilation/linking.